packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1011,7 +1054,39 @@ export class KeyringController extends BaseControllerV2<
    */
   async #addQRKeyring(): Promise<QRKeyring> {
     // QRKeyring is not yet compatible with Keyring type from @metamask/utils
-    return this.#keyring.addNewKeyring(KeyringTypes.qr) as unknown as QRKeyring;
+    const qrKeyring = (await this.#keyring.addNewKeyring(
+      KeyringTypes.qr,
+    )) as unknown as QRKeyring;
+
+    this.#subscribeToQRKeyringEvents(qrKeyring);
+
+    return qrKeyring;
+  }
+
+  /**
+   * Subscribe to a QRKeyring state change events and
+   * forward them through the messaging system.
+   *
+   * @param qrKeyring - The QRKeyring instance to subscribe to
+   */
+  #subscribeToQRKeyringEvents(qrKeyring: QRKeyring) {
+    this.#qrKeyringStateListener = (state) => {
+      this.messagingSystem.publish(`${name}:qrKeyringStateChange`, state);
+    };
+
+    qrKeyring.getMemStore().subscribe(this.#qrKeyringStateListener);
+  }
+
+  #unsubscribeFromQRKeyringsEvents() {
+    const qrKeyrings = this.#keyring.getKeyringsByType(
+      KeyringTypes.qr,
+    ) as unknown as QRKeyring[];
+
+    qrKeyrings.forEach((qrKeyring) => {
+      if (this.#qrKeyringStateListener) {
+        qrKeyring.getMemStore().unsubscribe(this.#qrKeyringStateListener);
+      }
+    });
```

Comment:
We shouldn't assume that we have just one `QRKeyring` as calling twice this expression:
```typescript
addNewKeyring(KeyringTypes.qr)
```
would end up creating two different `QRKeyring` instances

Reply:
The real problem with having multiple `QRKeyring` instances would be that the event listeners would publish through the same messenger event, making impossible from a client to establish which specific `QRKeyring` emitted that state update.

Reply:
A possible solution for this would be to subscribe to `QRKeyring` during `addNewAccount` **only** for the first keyring added: this would force a client that wants to use more than one `QRKeyring` instance to implement a subscribing solution on its side.

Reply:
Or, as we don't _explicitly_ support multiple keyrings of the same type (yet), we could simply throw an error when calling `addNewKeyring` if there's a keyring of that type already (with the exception of `SimpleKeyring`, but that one is created through `importAccountWithStrategy`).

Reply:
Ops, I actually forgot that I added this in `addNewKeyring` to prevent exactly this 🤦 :

```typescript
if (type === KeyringTypes.qr) {
   return this.getOrAddQRKeyring();
}
```

So the behavior in this PR is that if the `addNewKeyring` is called with a QR type, then the QRKeyring is retrieved or created